kickstarter/android-oss
# MBL-666: [Android] Update purchase event with new mutation on thanks page after user completes checkout

# 📲 What

- Created a `thirdPartyEventUseCase` to handle transforming and sending data to the `triggerThirdPartyEvent` graph mutation
- Replaced the CAPI purchase event for third party purchase event on the thanks screen after a user has backed a project
- Updated tests

# 🤔 Why

Adding support for GA4

# 📋 QA

- Make sure either the consent management feature flag and either the ga4 or CAPI/Meta feature flags are enabled in remote config on the internal debug variant.
- Run app in internal debug variant
- Find a project with `sendThirdPartyEvent = true`
- Consent management must be enabled/allowed on your device

<img width="1714" alt="Screenshot 2023-05-30 at 1 09 46 PM" src="https://github.com/kickstarter/android-oss/assets/19390326/3d1d47df-69e4-4a32-8df4-b34845f28d16">
<img width="1178" alt="Screenshot 2023-05-30 at 1 10 06 PM" src="https://github.com/kickstarter/android-oss/assets/19390326/e31cd631-136b-41b7-b184-19933247f171">


# Story 📖

[MBL-666: [Android] Update purchase event with new mutation on thanks page after user completes checkout](https://kickstarter.atlassian.net/browse/MBL-666)
